pkg/ibmvpc/nacl_analysis.go

```diff
@@ -74,23 +75,31 @@ func (na *IBMNACLAnalyzer) GetNACLRule(index int) (ruleStr string, ruleRes *comm
 		)
 		srcPorts := getPortsStr(*ruleObj.SourcePortMin, *ruleObj.SourcePortMax)
 		dstPorts := getPortsStr(*ruleObj.DestinationPortMin, *ruleObj.DestinationPortMax)
-		connStr = fmt.Sprintf("protocol: %s, srcPorts: %s, dstPorts: %s", *ruleObj.Protocol, srcPorts, dstPorts)
+		protocol = *ruleObj.Protocol
+		portsStr = fmt.Sprintf(", srcPorts: %s, dstPorts: %s", srcPorts, dstPorts)
 		direction = *ruleObj.Direction
 		src = *ruleObj.Source
 		dst = *ruleObj.Destination
 		action = *ruleObj.Action
 	case *vpc1.NetworkACLRuleItemNetworkACLRuleProtocolIcmp:
 		name = *ruleObj.Name
 		conns = commonvpc.GetICMPconn(ruleObj.Type, ruleObj.Code)
-		connStr = fmt.Sprintf("protocol: %s", *ruleObj.Protocol)
+		protocol = *ruleObj.Protocol
 		direction = *ruleObj.Direction
 		src = *ruleObj.Source
 		dst = *ruleObj.Destination
 		action = *ruleObj.Action
+		if ruleObj.Type != nil && *ruleObj.Type != -1 {
+			portsStr = fmt.Sprintf("icmp type: %d", *ruleObj.Type)
+		}
+		if ruleObj.Code != nil && *ruleObj.Code != -1 {
+			portsStr += fmt.Sprintf("icmp code: %d", *ruleObj.Code)
+		}
```

Comment:
```suggestion
		if ruleObj.Type != nil {
			portsStr = fmt.Sprintf(", type: %d", *ruleObj.Type)
		}
		if ruleObj.Code != nil {
			portsStr += fmt.Sprintf(", code: %d", *ruleObj.Code)
		}
```
The -1 value is an AWS-only thing